app/components/Header/DropdownContainer/FadeContents.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import React, { forwardRef } from 'react';
+import type { Node } from 'react';
+import styled, { keyframes } from 'styled-components';
+import styles from '../Header.css';
+
+const getFadeContainerKeyFrame = ({ animatingOut, direction }) => {
+  if (!direction) return;
+  return keyframes`
+  to {
+    transform: translateX(0px);
+    opacity: ${animatingOut ? 0 : 1};
+  }
+`;
+};
+
+const FadeContainer = styled.div`
+  animation-name: ${getFadeContainerKeyFrame};
```

Comment:
How does this work? Calling `getFadeContainerKeyFrame` without `{direction}` will `return;` so what is the point here? Is the `${getFadeContainerKeyFrame}` automatically called the prop values that match
```
      animatingOut={animatingOut}
      direction={direction}
```

Reply:
Because in the next line you use the props directly, which is a behavior I understand.
```
  opacity: ${(props) => (props.direction && !props.animatingOut ? 0 : 1)};
  ```

Reply:
I see that more code have this same behavior 🙃 , so I guess that's how it works. I kinda h8 that thooo, but okey 🥇 

One has to read three things in order to understand what values are being used:
1) This line `animation-name: ${getFadeContainerKeyFrame};`
2) This line `const getFadeContainerKeyFrame = ({ animatingOut, direction }) => {`
3) These lines
```
    <FadeContainer
      className={styles.fadeContainer}
      // prevent screen readers from reading out hidden content
      aria-hidden={animatingOut}
      animatingOut={animatingOut}
      direction={direction}
      ref={ref}
    >
 ```

Who figured that would be a good design pattern?